yt/data_objects/data_containers.py

```diff
@@ -772,35 +769,35 @@ def create_firefly_object(
             >>> region = ramses_ds.sphere(ramses_ds.domain_center, (1000, "kpc"))
 
             >>> reader = region.create_firefly_object(
-            ...     path_to_firefly="/Users/agurvich/research/repos/Firefly",
+            ...     "IsoGalaxyRamses",
             ...     fields_to_include=[
             ...         "particle_extra_field_1",
             ...         "particle_extra_field_2",
             ...     ],
             ...     fields_units=["dimensionless", "dimensionless"],
-            ...     dataset_name="IsoGalaxyRamses",
             ... )
 
             >>> reader.options["color"]["io"] = [1, 1, 0, 1]
             >>> reader.particleGroups[0].decimation_factor = 100
             >>> reader.dumpToJSON()
         """
 
-        ## attempt to import firefly_api
+        ## attempt to import Firefly
         try:
-            from firefly_api.particlegroup import ParticleGroup
-            from firefly_api.reader import Reader
+            from Firefly.data_reader import ParticleGroup, Reader
         except ImportError as e:
             raise ImportError(
-                "Can't find firefly_api, ensure it "
+                "Can't find Firefly, ensure it "
                 "is in your python path or install it with "
-                "`python -m pip install firefly_api`. It is also available "
-                "on github at github.com/agurvich/firefly_api"
+                "`python -m pip install Firefly-vis`. It is also available "
+                "on github at github.com/ageller/Firefly"
             ) from e
```

Comment:
I know this isn't firefly-specific and was already there, but we have a way of lazy-importing packages in yt (`yt/utilities/on_demand_imports.py`). This block of code should probably be moved there.

Reply:
oh for sure, i did not realize that existed. much much cleaner now